.github/workflows/auto-merge-abort.yml

```diff
@@ -0,0 +1,27 @@
+# This action aborts auto-merging by removing the label
+# when changes are pushed.
```

Comment:
I'm worried about TOCTOU problems here - it seems hard to "prove" for sure that they don't exist with how this is currently architectured. If you're confident that this implementation doesn't have TOCTOU problems, I'd thoroughly document it somewhere.

Reply:
In the github API, you can specify the sha hash to merge: https://docs.github.com/en/free-pro-team@latest/rest/pulls/pulls?apiVersion=2022-11-28#merge-a-pull-request
This means you can make sure that the PR was not updated with newer content than what your current state was.